system/jlib/jstats.cpp

```diff
@@ -775,190 +775,194 @@ class StatisticMeta
-    { WHENLASTSTAT(GraphFinished) }, // Deprecated - use WhenFinished
-    { WHENFIRSTSTAT(FirstRow) },
-    { WHENFIRSTSTAT(QueryStarted) }, // Deprecated - use WhenStart
-    { WHENLASTSTAT(QueryFinished) }, // Deprecated - use WhenFinished
-    { WHENFIRSTSTAT(Created) },
-    { WHENFIRSTSTAT(Compiled) },
-    { WHENFIRSTSTAT(WorkunitModified) },
-    { TIMESTAT(Elapsed) },
-    { TIMESTAT(LocalExecute) },
-    { TIMESTAT(TotalExecute) },
-    { TIMESTAT(Remaining) },
-    { SIZESTAT(GeneratedCpp) },
-    { SIZESTAT(PeakMemory) },
-    { SIZESTAT(MaxRowSize) },
-    { NUMSTAT(RowsProcessed) },
-    { NUMSTAT(Slaves) },
-    { NUMSTAT(Starts) },
-    { NUMSTAT(Stops) },
-    { NUMSTAT(IndexSeeks) },
-    { NUMSTAT(IndexScans) },
-    { NUMSTAT(IndexWildSeeks) },
-    { NUMSTAT(IndexSkips) },
-    { NUMSTAT(IndexNullSkips) },
-    { NUMSTAT(IndexMerges) },
-    { NUMSTAT(IndexMergeCompares) },
-    { NUMSTAT(PreFiltered) },
-    { NUMSTAT(PostFiltered) },
-    { NUMSTAT(BlobCacheHits) },
-    { NUMSTAT(LeafCacheHits) },
-    { NUMSTAT(NodeCacheHits) },
-    { NUMSTAT(BlobCacheAdds) },
-    { NUMSTAT(LeafCacheAdds) },
-    { NUMSTAT(NodeCacheAdds) },
-    { NUMSTAT(PreloadCacheHits) },
-    { NUMSTAT(PreloadCacheAdds) },
-    { NUMSTAT(ServerCacheHits) },
-    { NUMSTAT(IndexAccepted) },
-    { NUMSTAT(IndexRejected) },
-    { NUMSTAT(AtmostTriggered) },
-    { NUMSTAT(DiskSeeks) },
-    { NUMSTAT(Iterations) },
-    { LOADSTAT(WhileSorting) },
-    { NUMSTAT(LeftRows) },
-    { NUMSTAT(RightRows) },
-    { PERSTAT(Replicated) },
-    { NUMSTAT(DiskRowsRead) },
-    { NUMSTAT(IndexRowsRead) },
-    { NUMSTAT(DiskAccepted) },
-    { NUMSTAT(DiskRejected) },
-    { TIMESTAT(Soapcall) },
-    { TIMESTAT(FirstExecute) },
-    { TIMESTAT(DiskReadIO) },
-    { TIMESTAT(DiskWriteIO) },
-    { SIZESTAT(DiskRead) },
-    { SIZESTAT(DiskWrite) },
+    { StKindNone, SMeasureNone, StatsMergeSum, StKindNone, StKindNone, { "none" }, { "@none" }, nullptr },
+    { StKindAll, SMeasureAll, StatsMergeSum, StKindAll, StKindAll, { "all" }, { "@all" }, nullptr },
+    { WHENFIRSTSTAT(GraphStarted), "The time when a graph started./nDeprecated" }, // Deprecated - use WhenStart
+    { WHENLASTSTAT(GraphFinished), "The time when a graph finished./nDeprecated"  }, // Deprecated - use WhenFinished
+    { WHENFIRSTSTAT(FirstRow), "The time when the first row is processed by an activity"  },
+    { WHENFIRSTSTAT(QueryStarted), "The time when a query started./nDeprecated" }, // Deprecated - use WhenStart
+    { WHENLASTSTAT(QueryFinished), "The time when a query finished./nDeprecated" }, // Deprecated - use WhenFinished
+    { WHENFIRSTSTAT(Created), "The time when an item was created" },
+    { WHENFIRSTSTAT(Compiled), "The time a workunit started being compiled" },
+    { WHENFIRSTSTAT(WorkunitModified), UNUSED },
+    { TIMESTAT(Elapsed), "The elapsed time between starting and finishing\nFor child queries this may be significantly larger than TimeTotalExecute" },
```

Comment:
I think I was aiming for no fullstops since these are likely to appear in tooltips (so I would remove the ones above).  @GordonSmith what would you prefer?